redeem/gcodes/M2x.py

```diff
@@ -118,10 +118,12 @@ def execute(self, g):
             return
 
         # list all files on the device
-        # self.printer.send_message(g.prot, "files on external memory '{}'".format(list_location))
+        self.printer.send_message(g.prot, "Begin file list:")
         for root, directories, filenames in os.walk(list_location):
             for filename in filenames:
-                self.printer.send_message(g.prot, "{}/{} -".format(list_location, filename))
+                file_byte_count = os.stat(root + os.sep + filename).st_size
+                self.printer.send_message(g.prot, "{}/{} {}".format(device_id, filename, file_line_count))
```

Comment:
i think `file_line_count` should be `file_byte_count` (?)

Reply:
file_line_count iirc sends back the number of lines in a file - the number of bytes (file size) is what's given here in file_byte_count. I ran into a disconnect where I was at 0% and the print had ended... because there were 13k lines, but 130KB in the file or something like it, and it yielded the wrong percentage of progress in OP...